### PR TITLE
Added support for src port iteration

### PR DESCRIPTION
**Pull request by h3ct0r**
Hi,

This patch adds a new parameter (-i or --iterate-sport) to allow the iteration of the src ports instead of the destination port. I did run the test, and all seems ok! Not sure if I'm following the correct coding conventions so if the request need any modification just let me know!

Regards,

### CODE REVIEW

**Comment by insomniacslk** [comment]
hey @h3ct0r , apologies for the delay! Business travels and oncall, and I lost track of this. Reviewing right now :-)

**Comment by insomniacslk** [comment]
Thanks for your contribution! The code looks good to me.

There are just a couple minor nits. Additionally, this change breaks the interface, so we have to cut a new release, i.e. 0.5 . I will do that in a separate commit, and reflect on the Python module just after.

Additionally we want the same change in the Go version of this code, I've created https://github.com/insomniacslk/dublin-traceroute/issues/35 for the purpose

File: src/main.cc [hunk 1]
```diff
@@ -61,6 +63,7 @@ R"(Written by Andrea Barberio - https://insomniac.slackware.it
   -T MAX_TTL --max-ttl=MAX_TTL  the maximum TTL to probe. Must be greater or equal than the minimum TTL (default: )" << static_cast<int>(DublinTraceroute::default_max_ttl) << R"()
   -D DELAY --delay=DELAY        the inter-packet delay in milliseconds (default: )" << DublinTraceroute::default_delay << R"()
   -b --broken-nat               the network has a broken NAT configuration (e.g. no payload fixup). Try this if you see less hops than expected
+  -i --iterate-sport            iterate the source port instead of the destination port
```
Comment (insomniacslk) [inline src/main.cc]: For clarity can you change this to "Generate paths using source port instead of destination port" ?

**Comment by h3ct0r** [comment]
Hi @insomniacslk ! Yes sure, just give me a few days to make the changes :)

File: include/dublintraceroute/dublin_traceroute.h [hunk 2]
```diff
@@ -48,7 +48,8 @@ class  DublinTraceroute {
 				 min_ttl_,
 				 max_ttl_;
 	const uint16_t		 delay_;
-	const bool		 broken_nat_;
+	const bool		 broken_nat_,
+					iterate_sport_;
```
[the remark this answers is not in the document]
Comment (h3ct0r) [reply include/dublintraceroute/dublin_traceroute.h]: Yes sure, fixed!

File: include/dublintraceroute/dublin_traceroute.h [hunk 3]
```diff
@@ -63,6 +64,7 @@ class  DublinTraceroute {
 	static const uint8_t	 default_max_ttl = 30;
 	static const uint16_t	 default_delay = 10;
 	static const bool	 default_broken_nat = false;
+	static const bool	 default_iterate_sport = false;
```
[the remark this answers is not in the document]
Comment (h3ct0r) [reply include/dublintraceroute/dublin_traceroute.h]: I agree, now I have updated the variable to `use_srcport_for_path_generation`, it's clear and easily understandable!

File: src/main.cc (hunk 1 above)
Comment (h3ct0r) [reply src/main.cc]: Fixed!

**Comment by h3ct0r** [comment]
Hi @insomniacslk sorry for the delay, I have updated the pull request with the changes, let me know if you need any more changes! :)

**Comment by insomniacslk** [comment]
Thanks @h3ct0r, this looks good :) I will run some test locally to make sure the packet matching and the JSON generation are still correct before merging

**Comment by insomniacslk** [comment]
Alright - the matching is correct and the JSON output is good too. The Python module needs to be updated though, as it still references to the destination port in the plotted diagram, instead of the source port. I will comment on https://github.com/insomniacslk/python-dublin-traceroute/pull/12

**Pull request merged**